specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,741 @@
+
+ “protocols” : "[]string"
+
+ “createdAt” : “2019-03-20T12:49:00.497Z”
+
+ “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+ “name” : “string”
+
+ “description” : “string”
+ 
+ "shareId" : "string"
+
+ “shareSize” : ”int64”
+
+ “snapshotSize” : “int64”
+
+ “status”: “string”
+```
+##### 7.	GET /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Show details of snapshot
+##### Request
+```json
+  “id”: “string”
+```  
+##### Response parameters
+```json
+  “id”: “string”
+  
+  “shareId” : “string“
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 8.   POST /v1beta/{tenantId}/file/snapshot
+* Creates a snapshot from a share.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+
+  “shareId” : “string“
+```
+  ##### Response parameters
+```json  
+  “id”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 9.	PUT /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Updates a share snapshot.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+```  
+##### Response parameters
+```json
+  “id”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 10.	DELETE /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Deletes a share
+##### Request
+```json
+  “id” : “sting”
+```
+##### 11.	POST /v1beta/{tenantId}/file/accesses/
+* Add access control for file share.
+##### Request
+```json
+    “sharetId” : “sting”
+
+    “type” : “string”    // user/ip
+
+    “accessCapability” : read/write/execute
+
+    “accessTo” : “user_list”
+```
+##### 12.	PUT /v1beta/{tenantId}/file/accesses/{accessId}
+* Add access control for file share.
+##### Request
+```json
+   “id” : “string”
+```
+## Implementation of file share APIs for Profiles consolidated with block storage
+Note : Highlighted with background color texts are specific to file share and those are only supported in file share. Rest all parameters are belongs to block storage
+
+##### 1.	POST /v1beta/{tenantId}/profiles
+* Create profiles
+
+##### Request
+```json
+body*
+   [
+     ""name"         : "string",
+     "storage type" : "string",
+     "description"  : "string",
+     "provisioningProperties" : {
+                "datastore""  : {
+
+                   ""characterCodeSet"        : "string",
+                    "maxFileNameLengthBytes"  : "int64",
```

Comment:
Could it be possible to group the file share related parameters to File object?

Reply:
It will impact the block profile APIs, so right now we are not doing it.